lib/remap-lang/src/test_util.rs

```diff
@@ -16,3 +16,79 @@ macro_rules! test_type_def {
         }
     };
 }
+
+#[macro_export]
+macro_rules! func_args {
+    () => (
+        ::std::collections::HashMap::<&'static str, $crate::function::Argument>::default()
+    );
+    ($($k:tt: $v:expr),+ $(,)?) => {
+        vec![$((stringify!($k), $v.into())),+]
+            .into_iter()
+            .collect::<::std::collections::HashMap<&'static str, $crate::function::Argument>>()
+    };
+}
+
+#[macro_export]
+macro_rules! bench_function {
+    ($name:tt => $func:path; $($case:ident { args: $args:expr, want: $(Ok($ok:expr))? $(Err($err:expr))? $(,)* })+) => {
```

Comment:
Note the `want: $(Ok($ok:expr))? $(Err($err:expr))?` part.

This is needed because the compiler can't infer the type of `T` or `E` for `Result<T, E>` if only one of the two is defined. We also can't _provide_ a value for it because we don't know the type of `T` ahead of time.

An alternative would be to allow providing the type annotation when writing the benchmark or test for the function, but this avoids having to clutter the implementation code with those annotations.

TL;DR: this ain't pretty, but it works, and makes the actual implementations nicer to read.